assets/js/modules/tagmanager/components/common/ContainerNameTextField.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * ContainerNameTextField component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME, FORM_SETUP } from '../../datastore/constants';
+import { STORE_NAME as CORE_FORMS } from '../../../../googlesitekit/datastore/forms/constants';
+import { TextField, HelperText, Input } from '../../../../material-components';
+import { isUniqueContainerName } from '../../util';
+const { useSelect, useDispatch } = Data;
+
+export default function ContainerNameTextField( { label, name } ) {
+	const containers = useSelect( ( select ) => {
+		const accountID = select( STORE_NAME ).getAccountID();
+		return select( STORE_NAME ).getContainers( accountID );
+	} );
+	const containerName = useSelect( ( select ) => select( CORE_FORMS ).getValue( FORM_SETUP, name ) );
+
+	const { setValues } = useDispatch( CORE_FORMS );
+	const onChange = useCallback( ( { currentTarget } ) => {
+		setValues( FORM_SETUP, { [ name ]: currentTarget.value } );
+	}, [ name ] );
+
+	const helperText = containerName && ! isUniqueContainerName( containerName, containers )
+		? <HelperText persistent>{ __( 'A container with this name already exists.', 'google-site-kit' ) }</HelperText>
+		: undefined;
+
+	const trailingIcon = helperText
+		? <span className="googlesitekit-text-field-icon--error" />
+		: undefined;
+
+	return (
+		<div className="googlesitekit-tagmanager-containername">
+			<TextField
```

Comment:
Let's also add the `mdc-text-field--error` class if `!containerName || ! isUniqueContainerName`

Reply:
Added.